### ISSUE TITLE
depends system doesn't work for cross compilation from Ubuntu to Windows

### ISSUE THREAD

**Issue by TheHashCoin-project** [duality-solutions/Sequence#130]
Have tried using multiple versions of Ubuntu & Debian, but have not been able to successfully cross compile from Ubuntu/Debian to Windows (i686 or x86_64) using the depends system.

**Comment by SvenKercher** [comment]
We are using the official Gitian way, and there all files must be available at github, and we have no problems to compile from Ubuntu to Linux/Windows/OSX/ARM x86 and x64
So it is not the source that makes the problems

Im also able to do a direct cross compile from source without gitian

**Comment by TheHashCoin-project** [comment]
I haven't tried a Gitian build, but I can show you the output when I try a direct cross compile with the depends system. I know I am not missing dependencies, but perhaps one of the sources for the dependencies of the depends build are outdated, I didn't watch it the entire time to see if every dependency it fetches does so successfully. To build it regularly with autoconf (./autogen.sh, ./configure, make) on any Ubuntu distro works fine.

I am trying a fresh fetch and cross compile with the depends system now, I will paste the output when it's finished.

**Comment by TheHashCoin-project** [comment]
When you build with gitian, do you use lxc, or kvm?

**Comment by Duality-CDOO** [comment]
I believe he uses KVM.

**Comment by TheHashCoin-project** [comment]
Thank you! That's what I figured as I have not had great luck with LXC in the past. Will give it a shot.

**Comment by SvenKercher** [comment]
U can take a look at the gitian-win.yml file at contrib/gitian-descriptor , there u see the depends that u have to install at a direct compiler
After that u have to  cross compile all depends at the depends folder, how to is in the readme there
And then u can compile the binaries directly 

And yes, at gitian i use KVM

**Comment by AmirAbrams** [comment]
@TheHashCoin-project , did you have time to try to build with depends again?  I am going to close this issue and if you still can't cross compile, we can re-open it.  @SvenKercher would not be able to use gitian or build the Windows version if there was an issue.

**Comment by TheHashCoin-project** [comment]
I believe the issue was with the machine I was using for cross compilation,
it is very old and has had some memory/bus related issues and is a very
picky old beast haha. Thank you for your prompt attention to the issue!

**Comment by ultra-pool** [comment]
Iv tried to jack with gitian KVM & LXC for a week and its more worthless then a tick on a dog butt. haha. 
Cross compiling from depends doesnt work in this source. It has memory errors. I recommend looking into getting that fixed then relying on gitian for it.

**Comment by ultra-pool** [comment]
A buddy and I was able to fix the memory problem with cross compiling the source. Ill post changes in here after it fully working.

**Comment by Duality-CDOO** [comment]
@ultra-pool if you are going to open a pull request, please pr to the v1.3-WIP branch.

**Comment by ultra-pool** [comment]
@Duality-CDOO Github would not let me push the pr. I committed in v1.3-WIP the lines that i edited to be able to cross compile on Ubuntu without memory errors. I tested 3 times cross compiling and the cross compile worked each time.

**Comment by Duality-CDOO** [comment]
I would need to look into whether that would cause issues with building normally from source, have you tested @ultra-pool. Is that an unused function?

**Comment by ultra-pool** [comment]
Yes iv crossed compiled SEQ  3 times. All the SEQ functionality needs to be tested. 

All functions works on my forked source.

**Comment by SvenKercher** [comment]
The thing is, that i have no problems using gitian to compile all binaries
No problems at v1.3-WIP and no problems at master branch

**Comment by ultra-pool** [comment]
You would need to test compile with gitian. 

Me personally dont use gitian and never been able to get gitian to work correctly at all. Also compiling with gitian is a waste of space and time when you can cross compile without it in no time. 

But I posted what i did to be able to cross compile without gitian and seems to work in my forked source and cross compiling SEQ source.

**Comment by SvenKercher** [comment]
Cross compile without gitian is no option for a Release
For ur own usage u can do it without, but thats only personal usage
And for my System its no problem to handle the workload

**Comment by ultra-pool** [comment]
Have you tested the edit with gitian?
